presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -374,28 +374,37 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                             .collect(Collectors.toMap(key -> key, Symbol::toQualifiedNameReference)));
```

Comment:
We also need to change line 361 here so that node.getFilter is not propagated to the JoinNode now that the join predicate includes the filter from your changes.